python/ray/air/train/integrations/huggingface/huggingface_trainer.py

```diff
@@ -240,12 +240,12 @@ def trainer_init_per_worker(train_dataset, eval_dataset, **config):
     _dataset_config = {
         "train": DatasetConfig(
             required=True,
-            streamable=False,
-            _noncustomizable_fields=["streamable"],
+            use_stream_api=False,
+            _noncustomizable_fields=["use_stream_api"],
```

Comment:
@gjoliver what do you think about changing this to support an arbitrary validation function, like

```
_validator=DisallowUse(["use_stream_api"]),
```